hydrolib/core/io/inifield/models.py

```diff
@@ -0,0 +1,244 @@
+import logging
+from abc import ABC, abstractclassmethod
+from enum import Enum
+from pathlib import Path
+from typing import List, Literal, Optional
+
+from pydantic import Field
+from pydantic.class_validators import validator
+from pydantic.types import NonNegativeFloat, PositiveInt
+
+from hydrolib.core.io.ini.models import INIBasedModel, INIGeneral, INIModel
+from hydrolib.core.io.ini.util import (
+    get_enum_validator,
+    get_split_string_on_delimiter_validator,
+    make_list_validator,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DataFileType(str, Enum):
+    """
+    Enum class containing the valid values for the dataFileType
+    attribute in several subclasses of AbstractIniField.
+    """
+
+    arcinfo = "arcinfo"
+    geotiff = "GeoTIFF"
+    sample = "sample"
+    onedfield = "1dField"
+    polygon = "polygon"
+
+    allowedvaluestext = "Possible values: arcinfo, GeoTIFF, sample, 1dField, polygon."
+
+
+class InterpolationMethod(str, Enum):
+    """
+    Enum class containing the valid values for the dataFileType
+    attribute in several subclasses of AbstractIniField.
+    """
+
+    constant = "constant"  # only with dataFileType=polygon .
+    triangulation = "triangulation"  # Delaunay triangulation+linear interpolation.
+    averaging = "averaging"  # grid cell averaging.
+
+    allowedvaluestext = "Possible values: constant, triangulation, averaging."
+
+
+class Operand(str, Enum):
+    """
+    Enum class containing the valid values for the operand
```

Comment:
```suggestion
    Enum class containing the valid values for the interpolationMethod
```

Reply:
actually correct here, but you must have seen that it was wrong in class `InterpolationMethod`. I've fixed that.

Reply:
Oops sorry!